src/lib/mdns/DiscoveryManager.cpp

```diff
@@ -291,44 +313,100 @@ void DiscoveryManager::HandleNodeIdResolve(void * context, MdnsService * result,
     else
     {
         // Parse '%x-%x' from the name
-        uint64_t nodeId       = 0;
-        bool deliminatorFound = false;
+        uint64_t nodeId    = 0;
+        char * deliminator = nullptr;
+
+        HexStringToUint64(result->mName, sizeof(result->mName), &deliminator);
 
-        for (size_t i = 0; i < sizeof(result->mName) && result->mName[i] != 0; i++)
+        if (deliminator == nullptr || *deliminator != '-')
```

Comment:
can deliminator be nullptr?

also probably `delimitator` ?

Reply:
`delimiter`, please....

Reply:
delimiter?